### PR TITLE
[Verification] Cleans up the old verification architecture

### PR DESCRIPTION
**Pull request by yhassanzadeh13**
This PR removes the implementation of the old architecture of the verification node (mainly around finder and match engines), and refactors the metrics of the verification node to reflect the new engines' name.

### CODE REVIEW

File: engine/testutil/nodes.go [hunk 1]
```diff
@@ -626,201 +617,6 @@ func WithGenericNode(genericNode *testmock.GenericNode) VerificationOpt {
 	}
 }
 
-// TODO: this is fixture for old verification node (i.e., the one currently in place)
-// remove it once we have the new verification node rolled-in.
-func VerificationNode(t testing.TB,
```
Comment (yhassanzadeh13) [inline engine/testutil/nodes.go:631]: Removing the old factory for the verification node.

File: module/metrics/namespaces.go [hunk 2]
```diff
@@ -40,8 +40,9 @@ const (
 
 // Consensus subsystems represent the different components of the consensus algorithm.
 const (
-	subsystemCompliance = "compliance"
-	subsystemHotstuff   = "hotstuff"
+	subsystemCompliance  = "compliance"
+	subsystemHotstuff    = "hotstuff"
+	subsystemMatchEngine = "match"
```
Comment (yhassanzadeh13) [inline module/metrics/namespaces.go:45]: The consensus node was using the `subsystemMatchEngine` from the verification node domain, now that we no longer have a match engine, this subsystem label is moved up to the consensus domain.

File: engine/testutil/nodes.go [hunk 3]
```diff
@@ -626,201 +617,6 @@ func WithGenericNode(genericNode *testmock.GenericNode) VerificationOpt {
...
-			processInterval)
-		require.Nil(t, err)
-	}
-
-	return node
-}
-
 // NewVerificationNode creates a verification node with all functional engines and actual modules for purpose of
 // (integration) testing.
 //
```
Comment (jordanschalm) [inline engine/testutil/nodes.go:622]: Should we rename this `VerificationNode` per comment on line 623/827?

**Review by jordanschalm — APPROVED** [review]
🧹

**Review by zhangchiqing — APPROVED** [review]
🧹

File: engine/testutil/nodes.go (hunk 3 above)
Comment (yhassanzadeh13) [reply engine/testutil/nodes.go]: fixed in https://github.com/onflow/flow-go/pull/831/commits/4ad4c77709101c6ea7105867a8b3090fdb20d17c

**Pull request merged**